src/tools/utils.rs

```diff
@@ -115,3 +119,53 @@ where
     );
     Ok(())
 }
+
+/// block on function
+pub(super) fn block_on<T, F: Future<Output = T>>(f: F) -> Result<T> {
+    Ok(tokio::runtime::Runtime::new()?.block_on(f))
+}
+
+async fn index_from_file<K>(
+    header: &IndexHeader,
+    path: &Path,
+) -> AnyResult<BTreeMap<Vec<u8>, Vec<record::Header>>>
+where
+    for<'a> K: Key<'a> + 'static
+{
+    let headers = match header.version() {
+        5 => block_on(async {
```

Comment:
Current `HEADER_VERSION` constant should be used here.

Also, the idea behind making this function `async` was to remove this `block_on` call from its body